common/browsered/field-validation.js

```diff
@@ -95,23 +98,42 @@ function applyErrorMessaging (form, field, result) {
   let formGroup = field.closest('.form-group')
   if (!formGroup.classList.contains('error')) {
     formGroup.classList.add('error')
-    let label = document.querySelector('label[for="' + field.name + '"]')
-    let errorLabel = label.getAttribute('data-error-label')
-    if (errorLabel) {
-      result = errorLabel
+    const errorLegendElement = formGroup.querySelector('legend')
```

Comment:
No need to set a const if you only use it once

Reply:
Sorry, I am not sure what you mean?
We set `const` because we do not assign variable twice (`const` restricts the initialisation to be done only once) as that is the case (we don't want `let` for `errorLegendElement` as we do not reassign it).
Please let me know if you mean something else? Thanks

Reply:
Sorry I was writing on my phone before so couldn’t write code easily I mean its not worth setting up a variable if you only use it once.

So could just do 

```js
if (formGroup.querySelector('legend') === null) {}
```

Reply:
Thanks @jonheslop , as discussed, we refactored the validation logic